### PR TITLE
fix: mobile autofill from native suggestion

### PR DESCRIPTION
**Pull request by hogiyogi**
- **What does this PR do?**
Provides a fix for mobile browsers to autofill codes directly from their native message notifications.
This should resolve this issue as well: https://github.com/devfolioco/react-otp-input/issues/322

- **Any background context you want to provide?**
Pretty much just used @weiner-rakuten 's solution from [this pull request](https://github.com/devfolioco/react-otp-input/pull/348) but on the latest version.

- **Screenshots and/or Live Demo**

iOS - iPhone 12
https://github.com/devfolioco/react-otp-input/assets/25204032/57081773-1a69-43f1-92cb-ca42165d1001

Android - Pixel 7
https://github.com/devfolioco/react-otp-input/assets/25204032/02eec490-e6a6-40e2-be98-6cf627eb59cc

### CODE REVIEW

**Comment by hogiyogi597** [comment]
I'm not sure if this is the appropriate protocol to get reviewers so I apologize in advance but I wondered if @prateek3255 or @cesarvargas00 would be available in the next few days to give this one a look?

Thanks!

**Review by alexlibak — APPROVED** [review]

**Comment by hogiyogi597** [comment]
Hello! Just wanting to bump this thread a bit. I know everyone is probably busy but I don't want these changes to get lost in the GitHub ether.

Tagging for visibility:
@prateek3255 @cesarvargas00

**Comment by beforeoafterm** [comment]
Hello! Our team is also hoping this gets fixed. Any chance the maintainers can look into this PR?

**Comment by mckaymichaelis** [comment]
Also hoping for a fix for this 🤞

**Comment by alexlibak** [comment]
Same here, waiting for the fix 🙏🏼

**Comment by hogiyogi597** [comment]
Hey, all! Still hoping to get this merged but need to fork the repo for now to get this issue resolved for us.

Please let me know how I can help, thanks!

CC:
@prateek3255 @apollonian

**Comment by avtarvikas** [comment]
@hogiyogi597 when will this be merged ?
any idea ?

File: src/index.tsx [hunk 1]
```diff
@@ -238,7 +251,6 @@ const OTPInput = ({
               onKeyDown: handleKeyDown,
               onPaste: handlePaste,
               autoComplete: 'off',
-              maxLength: 1,
```
Comment (prateek3255) [inline src/index.tsx:241]: Is it necessary to remove maxLength for this to work? Hoping it does not break other cases where you paste an arbitrary value

**Comment by prateek3255** [comment]
Hey @hogiyogi597 Sorry for the delay here I might have missed the notification for the PR. I just have one question rest looks good

File: src/index.tsx (hunk 1 above)
Comment (hogiyogi597) [reply src/index.tsx:241]: Yes, unfortunately with the `maxLength` of 1 it is truncating the rest of the input. Are there some scenarios you'd like me to test? I'd be happy to help in any way I can.

**Comment by hogiyogi597** [comment]
@prateek3255  Thanks for looking it over! I responded to your question above. Let me know if there is any additional testing you'd like me to perform.

**Review by prateek3255 — APPROVED** [review]

**Comment by prateek3255** [comment]
Hey @hogiyogi597 I tested it on my end as well and `isInputValid` handles all the edge cases by itself anyways so we can get rid of `maxLength`. This is released with v3.1.1. Again thanks for your contribution and sorry for the dealy

**Comment by prateek3255** [comment]
@all-contributors please add @hogiyogi597 for code

**Comment by alexlibak** [comment]
Great work @hogiyogi597 !
Just updated it at my work's repository project and autofill works with ios 💪🏼

**Pull request merged**